File: Math/2Solution.py
def countBits(n):
        bitArray = [-1]*(n+1)
        for i in range(n+1):
            binary = bin(i).replace("0b","")
            bitArray[i] = binary.count("1")
        return bitArray
n=int(input())
print(countBits(n))

File: Math/2Solution.java
class Solution {
    public int[] countBits(int n) {
        int[] arr=new int[n+1];
        for(int i=0;i<=n;i++){
            int num=i,count=0;
            while(num>0){
                if(num%2==1){
                    count++;
                }num/=2;
            }
            arr[i]=count;
        }
        return arr;
    }
}
